lms/validation/_lti.py

```diff
@@ -1,6 +1,134 @@
 from webargs import fields
+import marshmallow
+from marshmallow.validate import OneOf
 
-from lms.validation._base import PyramidRequestSchema
+from lms.validation._base import PyramidRequestSchema, Reach
+
+
+class LTIAuthParamsSchema(marshmallow.Schema):
+    class Meta:
+        unknown = marshmallow.EXCLUDE
+
+    user_id = marshmallow.fields.Str(required=True)
+    roles = marshmallow.fields.Str(required=True)
+    tool_consumer_instance_guid = marshmallow.fields.Str(required=True)
+    lis_person_name_given = marshmallow.fields.Str(load_default="")
+    lis_person_name_family = marshmallow.fields.Str(load_default="")
+    lis_person_name_full = marshmallow.fields.Str(load_default="")
+    lis_person_contact_email_primary = marshmallow.fields.Str(load_default="")
+
+
+class LTI13AuthParamsSchema(LTIAuthParamsSchema):
```

Comment:
This could be tidier up but the marshmallow approach as a few problems IMO:

- Marshmallow doesn't make it nicer that just a map of keys, arguably 

```
   tool_consumer_instance_guid = Reach(
        fields.Str(),
   data_key="https://purl.imsglobal.org/spec/lti/claim/tool_platform",
        path="guid",
        required=True,
)
```

is harder to read that the other solution.

- We need to do the mapping in each Schema with spreads the "LTI1.3 mapping problem" across multiple files.


-As is tied to the schema is not possible to use the mappings without one. This is a problem when we don't have a "parsed_params" and we still need to access the LTI parameters as we do in the predicates.